src/signer/src/main/java/ee/ria/xroad/signer/tokenmanager/token/SoftwareTokenWorker.java

```diff
@@ -197,6 +195,7 @@ protected void deleteCert(String certId) throws Exception {
         Signature signature = Signature.getInstance(SIGNATURE_ALGORITHM);
```

Comment:
Since you have introduced the signature algorithm as a parameter, maybe also the software token should fail if the argument is not supported (cf. HardwareTokenWorker).
 
It seems that the method expects a digest as the data (and at least BatchSigner seems to always calculate one), so it could check that the algorithm is one of SHA*withRSA (?).